app/main/decorators.py

```diff
@@ -2,59 +2,49 @@
 
 from flask import abort, current_app, g
 
-from app.main.authorisation import AuthBase, AuthMapping
+from app.main.authorisation import AuthBase
 from app.main.fund import FundConfig
 
 
-def auth_required(func):
-    @wraps(func)
-    def decorated_function(*args, **kwargs):
-        """Checks that the user is authorised to submit.
-
-        Checks if a user is authorised to submit by:
-            1. deriving the fund they're submitting for from their role
-            2. fetching the relevant auth mappings for that fund
-            3. retrieving the auth for that user from the relevant auth mapping
+class Access:
+    fund: FundConfig
+    auth: AuthBase
 
-        If the fund window is active and user is authorised to submit, two attributes are added to the request context:
-            - g.fund: contains fund specific context used throughout the application
-            - g.auth: contains information that determine what the user is allowed to submit
+    def __init__(self, fund: FundConfig, auth: AuthBase):
+        self.fund = fund
+        self.auth = auth
 
-        Otherwise, aborts and redirects to 401 (unauthorised) page.
 
-        TODO: As mentioned in app/__init__.py, going forwards we should look to extract and encapsulate this "fund" and
-            "auth" data in separate microservices.
-
-        :raises 401 Unauthorized: If the user has an invalid role(s) or no auth.
-        """
-        funds: list[FundConfig] = current_app.config["FUND_CONFIGS"].get_active_funds(g.user.roles)
-
-        if not funds:
-            current_app.logger.info(f"User: {g.user.email} is not linked with any active funds.")
+def set_user_access(func):
+    @wraps(func)
+    def decorated_function(*args, **kwargs):
+        available_funds = current_app.config["FUND_CONFIGS"].get_active_funds(g.user.roles)
+        access = {}
+        for fund in available_funds:
+            auth_mapping = current_app.config["AUTH_MAPPINGS"].get_auth(fund.fund_name)
+            auth = auth_mapping.get_auth(g.user.email)
+            if auth is None:
+                current_app.logger.info(f"User: {g.user.email} is not authorised to submit for fund: {fund.fund_name}")
+                continue
```

Comment:
Do we want to `continue` here? Would we not abort in that case? Maybe I'm missing something?

Reply:
The code did previously throw a 400 error, however it was modified to continue with the flow and simply skip adding the fund to the list on the front end, as per Cyrus' suggestion here https://github.com/communitiesuk/funding-service-design-post-award-submit/pull/141#discussion_r1516001766